util/chunk/column.go

```diff
@@ -608,7 +608,7 @@ func (c *Column) reconstruct(sel []int) {
 // CopyReconstruct copies this Column to dst and removes unselected rows.
 // If dst is nil, it creates a new Column and returns it.
 func (c *Column) CopyReconstruct(sel []int, dst *Column) *Column {
-	if sel == nil {
+	if sel == nil || len(sel) == c.length {
```

Comment:
This assumption that `sel` is ordered is useful and can bring much performance improvement for almost all expressions since this method will be called by all `Columns`.